File: Python/Recursions/remove_adj_duplicates.py
def remove_duplicate(str):
    if not str:
        return ""
    
    if len(str) == 1:
        return str
    
    if str[0] == str[1]:
        return remove_duplicate(str[1:])
    
    return str[0] + remove_duplicate(str[1:])

print(remove_duplicate("Hello Woorlld"))

File: JavaScript/Recursions/removeAdjDuplicates.js
const removeDuplicates = (str) => {
    if(!str) {
        return "";
    }
    if(str.length === 1) {
        return str;
    }
    if(str[0] === str[1]) {
        return removeDuplicates(str.slice(1));
    }
    return str[0] + removeDuplicates(str.slice(1));
}


console.log(removeDuplicates("Hello Worlldd"));